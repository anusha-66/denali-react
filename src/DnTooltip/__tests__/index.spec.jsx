/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnTooltip, DnTooltipPositions, DnLink, DnIcon } from '../../index'; // src/index.tsx

describe('DnTooltip Tests', () => {
    it('should export DnTooltipPositions', () => {
        expect.assertions(1);

        expect(DnTooltipPositions).toStrictEqual({
            default: 'default',
            top: 'top',
            right: 'right',
            left: 'left',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTooltip tooltipText="the tooltip text">
                <DnLink>Tooltip</DnLink>
            </DnTooltip>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <div className="">
                <div className="flex justify-content-center m-t-40">
                    <DnTooltip
                        className="additional--class"
                        position={DnTooltipPositions.right}
                        isSmall
                        tooltipText="the tooltip text"
                    >
                        <DnIcon icon="user-profile-circle" />
                    </DnTooltip>
                </div>
            </div>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
