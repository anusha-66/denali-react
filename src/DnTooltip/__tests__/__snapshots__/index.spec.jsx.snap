// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnTooltip Tests should render correctly with custom props 1`] = `
<div
  class=""
>
  <div
    class="flex justify-content-center m-t-40"
  >
    <span
      class="d-tooltip-right-small additional--class"
      data-tooltip="the tooltip text"
    >
      <i
        class="d-icon d-user-profile-circle is-default"
      />
    </span>
  </div>
</div>
`;

exports[`DnTooltip Tests should render with default props 1`] = `
<span
  class="d-tooltip-default"
  data-tooltip="the tooltip text"
>
  <span
    class="link"
  >
    Tooltip
  </span>
</span>
`;
